src/utils/daos_hdlr.c

```diff
@@ -2485,6 +2485,7 @@ fs_copy(struct file_dfs *src_file_dfs,
 				}
 				next_dpath = strdup(dst_filename);
 				if (next_dpath == NULL) {
+					D_FREE(next_path);
```

Comment:
@daltonbohning if no memory was allocated for next_path, should this instead be freeing the dst_filename?

Reply:
@dsikich `dst_filename` is cleaned up at the bottom of this function, so we should be okay there as long as we use D_GOTO.
I do think we should free `next_path`, because if `next_dpath` fails to allocate, we still have `next_path` allocated from above.
(Notice the difference between `path` and `dpath`.